src/CodeGen_HLS_Target.h

```diff
@@ -58,6 +58,8 @@ class CodeGen_HLS_Target {
 
         void visit(const For *op);
         void visit(const Allocate *op);
+    private:
+        std::string guess_name(const std::string &name, uint32_t index);
```

Comment:
This function should be a const static member function or static non-member function.